### DISCUSSION TITLE
[Q&A] GLPI agent shortname/fqdn discovery issue

### DISCUSSION THREAD

**Discussion by liedekef** [glpi-project/glpi-agent#339] [Q&A]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

When installing an agent on a server, the server gets added correctly to glpi, using the server hostname as "Name" in glpi.
The hostname on the server is the "short" hostname, which is ok, so for example "myservername" gets added to glpi, togethet with his network interfaces and ip info.
Now, when I do a network discovery, the server also gets detected, but:
- on the network, the servername is fully qualified
- so: the name of the asset changes to "myservername.mydomain" in glpi *and* the interfaces info gets removed.

After that, when the agent runs again, the name changes again back to "myservername" and the interfaces get added again.

So my question is a bit double here:
- should the "name" of the agent be the fqdn? Or make the detection more intelligent so the agent detects the domain too in case the hostname is the shortname?
- would it maybe be best to not discover servers at all (and only network equipment)? While we want discovery to be aware of new servers, changing the settings on existing servers is not desired here (not in this case anyway).

Franky

**Comment by g-bougard** [comment]
Hi @liedekef 
agent name are agent names. This may make trouble with your task handling in GlpiInventory if the agent name changes.
Honestly this is not normal the agent name changes the way you're explaining (maybe I'm misunderstanding you).
Can you explain in which context the agent is changing its name ? I mean:
 * what is the OS and OS version,
 * what is the agent and agent version,
 * how agent is installed ?
 * how agent is started ? Is it running as a service/daemon or scheduled with crontab or job scheduler ?
In general, if the agent name changes, this means it lost the agent name or deviceid it created the first time it started.
